src/main/java/com/cburch/logisim/file/XmlWriter.java

```diff
@@ -106,39 +97,36 @@ static String attrsToString(NamedNodeMap a) {
   }
 
   static int stringCompare(String a, String b) {
-    if (a.equals(b)) return 0;
-    else if (a == null) return -1;
-    else if (b == null) return 1;
-    else return a.compareTo(b);
+    if (a == null) return -1;
+    if (b == null) return 1;
+    return a.compareTo(b);
+  }
+  
+  static final void swap(Node top, Node nodeA, Node nodeB) {
+    final var na = nodeA.getNodeName();
+    final var nb = nodeB.getNodeName();
+    var compare = stringCompare(na, nb);
+    if (compare > 0) {
+      top.insertBefore(nodeB, nodeA);
+    } else if (compare < 0) return;
+    final var ma = attrsToString(nodeA.getAttributes());
+    final var mb = attrsToString(nodeB.getAttributes());
+    compare = stringCompare(ma, mb);
+    if (compare > 0) {
+      top.insertBefore(nodeB, nodeA);
+    } else if (compare < 0) return;
+    final var va = nodeA.getNodeValue();
+    final var vb = nodeB.getNodeValue();
+    compare = stringCompare(va, vb);
+    if (compare > 0) {
+      top.insertBefore(nodeB, nodeA);
+    }
   }
-
-  static final Comparator<Node> nodeComparator =
-      (a, b) -> {
-        final var na = a.getNodeName();
-        final var nb = b.getNodeName();
-        var c = stringCompare(na, nb);
-        if (c != 0) return c;
-        final var ma = attrsToString(a.getAttributes());
-        final var mb = attrsToString(b.getAttributes());
-        c = stringCompare(ma, mb);
-        if (c != 0) return c;
-        final var va = a.getNodeValue();
-        final var vb = b.getNodeValue();
-        c = stringCompare(va, vb);
-        return c;
-        // This can happen in some cases, e.g. two text components
-        // on top of each other. But it seems rare enough to not
-        // worry about, since our normalization here is just for
-        // ease of comparing circ files during testing.
-        // System.out.printf("sorts equal:\n");
-        // System.out.printf(" a: <%s %s>%s\n", na, ma, va);
-        // System.out.printf(" b: <%s %s>%s\n", nb, mb, vb);
-      };
 
   static void sort(Node top) {
-    NodeList children = top.getChildNodes();
-    int n = children.getLength();
-    String name = top.getNodeName();
+    final var children = top.getChildNodes();
+    final var nrChildren = children.getLength();
```

Comment:
What about renaming `nrChildren` to `childrenCount`? It'd nicely group related vars when IDE hints them.

Reply:
Maybe it requires a kind of guideline on variable naming, as this is very personal, another point for the roadmap discussion